SD-K9/src/app/components/outdoor-map/outdoor-map.component.ts

```diff
@@ -288,4 +300,29 @@ export class OutdoorMapComponent implements OnInit, OnDestroy, Map {
     this._unsubscribeGoogleStore();    // release listener to google-store
   }
 
+  
+
+  inCampus(coordinates){ 
+    //coordinates = new google.maps.LatLng(45.497307, -73.578971);   //to test Hall Building coordinates
+    let isInside = false;
+    let building = "";
+    let polygonArr: any[] = [];
+    polygonArr = this.buildingPolygons;
+    //verify if user location is inside any campus building
+    for(var i = 0; i < polygonArr.length; i++){
+      isInside = google.maps.geometry.poly.containsLocation(coordinates, polygonArr[i]);
+      if(isInside){
+        building = polygonArr[i].currentBuilding;
+        break;
+      }
+    }
+
+    if(isInside){
+      //display you are inside a campus building
+      document.getElementById('btn').innerHTML = "You are in " + building + " building";
```

Comment:
The angular way of doing this is using [Template syntax](https://angular.io/guide/template-syntax#interpolation-and-template-expressions). You can have a variable message = 'something' and just change the message, and in the HTML file, you would have <h3 id="btn">{{message}}</h3> so you don't need to deal with getting the element by id and changing the inner HTML